feat: defaultRegistryUrls (#13907)
renovatebot/renovate: docs/usage/configuration-options.md

This option applies only to the `gradle` manager.

> ## defaultRegistryUrls

Reviewer: I'm still a bit fuzzy on _why_ I would need this feature. What it the problem that `defaultRegistryUrls` solves? _When_ would I need to use this feature?

Author: I added some clarification below. Most people won't need to use this, so it's not important for all to understand

> This configuration option enables overriding of a datasource's default registries.

Reviewer: ```suggestion
Override the datasource's default registries with this config option.
```

How about something like this?

> Like `registryUrls`, it is only applicable when the datasource's `customRegistrySupport` value is set to `true`.

Reviewer: ```suggestion
The datasources's `customRegistrySupport` value must be `true` for the config option to work.
```

Default registries are only used when both:

> - The manager does not extract any `registryUrls` values, and

Reviewer: ```suggestion
- The manager did not extract any `registryUrls` values, and
```

The next list item uses the passive tense. So we should use `did` here to be consistent, I think.

- No `registryUrls` values have been applied via config, such as `packageRules`

> Think of `defaultRegistryUrls` as a way to specify the "fallback" registries for a datasource, where supported.

Author: ```suggestion
Think of `defaultRegistryUrls` as a way to specify the "fallback" registries for a datasource, for use when no `registryUrls` are extracted or configured.
Compare that to `registryUrls`, which are a way to _override_ registries.
```

> In other words, you can think of it as a way to specify the "fallback" registries for a datasource, where supported.

Reviewer: ```suggestion
Think of `defaultRegistryUrls` as a way to specify the "fallback" registries for a datasource, where supported.
```

We can strip the `In other words,` part.

I think `it` refers to the `defaultRegistryUrls` config option?

> In other words, you can think of it as a way to specify the "fallback" registries for a datasource, where supported.

Author: ```suggestion
In other words, you can think of it as a way to specify the "fallback" registries for a datasource, where supported.

Here is an example usage of this feature:

- You want to change the default Maven registry to be your own private registry, but
- You don't wish to force all projects/lookups to go through your private registry, if other registries are explicitly configured

In this case, using `defaultRegistryUrls` for the `maven` datasource would make sense, because using `registryUrls` would force override any registries found within the repository's `settings.xml` files.
```
